Packages/com.llealloo.audiolink/Runtime/Scripts/WebALPeer.cs

```diff
@@ -0,0 +1,42 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+#if UNITY_WEBGL
+namespace AudioLink
+{
+    public class WebALPeer : MonoBehaviour
+    {
+        const int SAMPLES_COUNT = 4096;
+        private float[] _waveformSamplesLeft, _waveformSamplesRight;
+
+        public WebALPeer()
+        {
+            _waveformSamplesLeft = new float[SAMPLES_COUNT];
+            _waveformSamplesRight = new float[SAMPLES_COUNT];
+        }
+
+        public float[] GetWaveformLeft()
+        {
+            return _waveformSamplesLeft;
+        }
+
+        public float[] GetWaveformRight()
+        {
+            return _waveformSamplesRight;
+        }
+
+        public void SyncLeft(Action<float[]> FetchSamplesLeft)
```

Comment:
why not just regular setters?

Reply:
idk I think fundale wrote that

Reply:
maybe change it? looks like a simple change to make

Reply:
@fundale any reason you wrote it like this?

Reply:
> @fundale any reason you wrote it like this?

Not particularly,
A setter might work, I just did it the same way as this shim:
https://github.com/Lakistein/Unity-Audio-Visualizers

Reply:
hmmm i tried today, my attempt didn't work [ce64511](https://github.com/llealloo/audiolink/pull/312/commits/ce645114a7b2399c02e1a8d14b0b88663f2143c6)

Reply:
```
            if (audioSource.isPlaying)
            {
                audioLinkWebPeer.WaveformSamplesLeft = FetchAnalyzerLeft(WebALID, new float[4096], 4096);
                audioLinkWebPeer.WaveformSamplesRight = FetchAnalyzerRight(WebALID, new float[4096], 4096);
            }
            
          ```
          should probably be 
      ```
                  if (audioSource.isPlaying)
            {
                FetchAnalyzerLeft(WebALID, audioLinkWebPeer.WaveformSamplesLeft, 4096);
                FetchAnalyzerRight(WebALID, audioLinkWebPeer.WaveformSamplesRight, 4096);
            }
      ```